test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,161 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/params"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+)
+
+// vmFakeintakeSuite defines a test suite for the log agent interacting with a virtual machine and fake intake.
+type vmFakeintakeSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+}
+
+// logsExampleStackDef returns the stack definition required for the log agent test suite.
+func logsExampleStackDef() *e2e.StackDefinition[e2e.FakeIntakeEnv] {
+	config :=
+		`logs:
+  - type: file
+    path: '/var/log/hello-world.log'
+    service: hello
+    source: custom_log
+`
+	return e2e.FakeIntakeStackDef(
+		e2e.WithAgentParams(
+			agentparams.WithLogs(),
+			agentparams.WithIntegration("custom_logs.d", config)))
+
+}
+
+// TestE2EVMFakeintakeSuite runs the E2E test suite for the log agent with a VM and fake intake.
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	e2e.Run(t, &vmFakeintakeSuite{}, logsExampleStackDef(), params.WithDevMode())
+}
+
+func (s *vmFakeintakeSuite) TestLinuxLogTailing() {
+	// Clean up once test is finished running
+	s.cleanUp()
+	defer s.cleanUp()
+
+	// Flush server and reset aggregators
+	s.Env().Fakeintake.FlushServerAndResetAggregators()
+	defer s.Env().Fakeintake.FlushServerAndResetAggregators()
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Have you considered calling these from a `BeforeTest` or `TearDownTest` --> https://pkg.go.dev/github.com/stretchr/testify/suite#TearDownSubTest

These are state management methods implemented by the testify TestSuite

Reply:
I did not know this exist, will apply it to the test!